hangman.py

```diff
@@ -1,213 +1,306 @@
-# Hint: You might want to pick your own secret_word while you're testing.
-
-
-if __name__ == "__main__":
-    # pass
-
-    # To test part 2, comment out the pass line above and
-    # uncomment the following two lines.
-    
-    secret_word = choose_word(wordlist)
-    hangman(secret_word)
-
-###############
-    
-    # To test part 3 re-comment out the above lines and 
-    # uncomment the following two lines. 
-    
-    #secret_word = choose_word(wordlist)
-    #hangman_with_hints(secret_word)
+# Problem Set 2, hangman.py
+# Name: Alexei Baida
+# Collaborators:
+# Time spent: 8 - 10 hours
+
+import random
+import string
+
+WORDLIST_FILENAME = "words.txt"
+guesses = 6
+
+
+class TextStyle:
+    """Class for styling text with different colour/font"""
+    def yellow(self, text):
+        print('\033[33m' + text + '\033[0m')
+
+    def bold(self, text):
+        print('\033[1m' + text + '\033[0m')
+
+    def red(self, text):
+        print('\033[31m' + text + '\033[0m')
+
+    def change_background_font(self, text):
+        print('\033[7m' + text + '\033[0m')
+
+
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print("Loading word list from file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # line: string
+    line = inFile.readline()
+    # wordlist: list of strings
+    wordlist = line.split()
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+def choose_word(wordlist):
+    """
+    wordlist (list): list of words (strings)
+
+    Returns a word from wordlist at random
+    """
+    return random.choice(wordlist)
+
+# Load the list of words into the variable wordlist
+# so that it can be accessed from anywhere in the program
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    # & means intersection of these 2 sets
+    if set(secret_word) == set(secret_word) & set(letters_guessed):
+        return True
+    else:
+        return False
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    word = []
+    for letter in secret_word:
+        if letter in letters_guessed:
+            word += f'{letter}'
+        else:
+            word += '_'

```

Comment:
I believe it makes sense to create a constant for that '_' symbol

Reply:
Ok